### PR TITLE
Fix no pokeballs log spamming

### LINKED ISSUE #3630: Trying to catch same pokemon over and over again
### Expected Behavior

Skip the pokemon. Continue moving to fort.

### Actual Behavior

Trying to catch the pokemon over and over again.
```
[22:32:06] A wild Nidoran M appeared! [CP 496] [Potential 0.44] [A/D/S 6/3/11]
[22:32:09] No usable pokeballs found!
[22:32:19] Skipping Recycling of Items. 155 space left in bag.
[22:32:19] Something rustles nearby!                                                                                             [22:32:20] A wild Nidoran M appeared! [CP 496] [Potential 0.44] [A/D/S 6/3/11]
[22:32:24] No usable pokeballs found!
[22:32:33] Skipping Recycling of Items. 155 space left in bag.
[22:32:33] Something rustles nearby!                                                                                             [22:32:34] A wild Nidoran M appeared! [CP 496] [Potential 0.44] [A/D/S 6/3/11]
[22:32:37] No usable pokeballs found!
[22:32:48] Skipping Recycling of Items. 155 space left in bag.
[22:32:48] Something rustles nearby!                                                                                             [22:32:50] A wild Nidoran M appeared! [CP 496] [Potential 0.44] [A/D/S 6/3/11]
[22:32:53] No usable pokeballs found!
[22:33:02] Skipping Recycling of Items. 155 space left in bag.
[22:33:02] Something rustles nearby!                                                                                             [22:33:03] A wild Nidoran M appeared! [CP 496] [Potential 0.44] [A/D/S 6/3/11]
[22:33:06] No usable pokeballs found!
[22:33:16] Skipping Recycling of Items. 155 space left in bag.
[22:33:16] Something rustles nearby!                                                                                             [22:33:17] A wild Nidoran M appeared! [CP 496] [Potential 0.44] [A/D/S 6/3/11]
[22:33:20] No usable pokeballs found!
[22:33:29] Skipping Recycling of Items. 155 space left in bag.
[22:33:29] Something rustles nearby!                                                                                             [22:33:31] A wild Nidoran M appeared! [CP 496] [Potential 0.44] [A/D/S 6/3/11]
[22:33:34] No usable pokeballs found!
[22:33:43] Skipping Recycling of Items. 155 space left in bag.
[22:33:43] Something rustles nearby!                                                                                             [22:33:44] A wild Nidoran M appeared! [CP 496] [Potential 0.44] [A/D/S 6/3/11]
[22:33:48] No usable pokeballs found!
[22:33:59] Skipping Recycling of Items. 155 space left in bag.
[22:33:59] Something rustles nearby!                                                                                             [22:34:00] A wild Nidoran M appeared! [CP 496] [Potential 0.44] [A/D/S 6/3/11]
[22:34:03] No usable pokeballs found!
[22:34:12] Skipping Recycling of Items. 155 space left in bag.
[22:34:12] Something rustles nearby!                                                                                             [22:34:18] A wild Nidoran M appeared! [CP 496] [Potential 0.44] [A/D/S 6/3/11]
[22:34:22] No usable pokeballs found!
[22:34:31] Skipping Recycling of Items. 155 space left in bag.
[22:34:31] Something rustles nearby!                                                                                             
[22:34:32] A wild Nidoran M appeared! [CP 496] [Potential 0.44] [A/D/S 6/3/11]
[22:34:35] No usable pokeballs found!
```

### Your config.json (remove your credentials and any other private info)
```json
{
	...
	"tasks": [
		{
			"type": "UpdateLiveStats",
			"config": {
				"min_interval": 60,
				"stats": [
					"uptime",
					"km_walked",
					"level_stats",
					"xp_earned",
					"xp_per_hour",
					"stops_visited",
					"pokemon_stats",
					"pokeballs_thrown",
					"stardust_earned",
					"highest_cp_pokemon",
					"most_perfect_pokemon"
				]
			}
		},
		{
			"type": "HandleSoftBan",
			"config": {
				"enabled": true
			}
		},
		{
			"type": "CollectLevelUpReward",
			"config": {
				"enabled": false
			}
		},
		{
			"type": "IncubateEggs",
			"config": {
				"enabled": true,
				"longer_eggs_first": true
			}
		},
		{
			"type": "TransferPokemon",
			"config": {
				"enabled": true
			}
		},
		{
			"type": "NicknamePokemon",
			"config": {
				"enabled": false,
				"nickname_template": "{iv_pct}_{iv_ads}"
			}
		},
		{
			"type": "EvolvePokemon",
			"config": {
				"enabled": false,
				"evolve_all": "none",
				"first_evolve_by": "cp",
				"evolve_above_cp": 500,
				"evolve_above_iv": 0.8,
				"logic": "and",
				"evolve_speed": 20,
				"use_lucky_egg": false
			}
		},
		{
			"type": "RecycleItems",
			"config": {
				"enabled": true,
				"min_empty_space": 10,
				"item_filter": {
					"Pokeball": {
						"keep": 150
					},
					"Greatball": {
						"keep": 150
					},
					"Ultraball": {
						"keep": 150
					},
					"Masterball": {
						"keep": 150
					},
					"Potion": {
						"keep": 0
					},
					"Super Potion": {
						"keep": 0
					},
					"Hyper Potion": {
						"keep": 0
					},
					"Max Potion": {
						"keep": 75
					},
					"Revive": {
						"keep": 10
					},
					"Max Revive": {
						"keep": 20
					},
					"Razz Berry": {
						"keep": 100
					}
				}
			}
		},
		{
			"type": "CatchVisiblePokemon",
			"config": {
				"enabled": true
			}
		},
		{
			"type": "CatchLuredPokemon",
			"config": {
				"enabled": true
			}
		},
		{
			"type": "SpinFort",
			"config": {
				"enabled": true
			}
		},
		{
			"type": "MoveToMapPokemon",
			"config": {
				"enabled": true,
				"address": "http://127.0.0.1:8080",
				"max_distance": 50000,
				"min_time": 60,
				"min_ball": 20,
				"prioritize_vips": true,
				"snipe": true,
				"snipe_high_prio_only": true,
				"snipe_high_prio_threshold": 600,
				"update_map": false,
				"mode": "distance",
				"catch": { ... }
			}
		},
		{
			"type": "MoveToFort",
			"config": {
				"enabled": true,
				"lure_attraction": true,
				"lure_max_distance": 2000
			}
		},
		{
			"type": "FollowSpiral",
			"config": {
				"enabled": true,
				"diameter": 600,
				"step_size": 70
			}
		}
	],
	"map_object_cache_time": 5,
	"forts": {
		"avoid_circles": true,
		"max_circle_size": 50
	},
	"websocket_server": false,
	"walk": 80,
	"action_wait_min": 1,
	"action_wait_max": 2,
	"debug": false,
	"test": false,
	"health_record": false,
	"location_cache": true,
	"distance_unit": "km",
	"reconnecting_timeout": 10,
	"catch_randomize_reticle_factor": 1,
	"catch_randomize_spin_factor": 1,
	"logging_color": true,
	...
}
```

### Other Information
OS: Ubuntu  16.04
Git Commit: 138b66413e45efcef1ae015388b80a2725e7a7d8
Python Version: 2.7.12

### PR DESCRIPTION
**Pull request by umbreon222**
## Short Description:
This fixes the no_pokeballs spam in console by checking pokeball count AoT. Also, fixes a possible bug in equidistant pokestops that could occur.

## Fixes (provide links to github issues if you can):
Closes #3630

### CODE REVIEW

**Comment by douglascamata** [comment]
@umbreon222 you will never have 2 pokestops equally distant from you, because precision is down to meters. The probability is this happening is super, super low. The no pokeball spam fix is very welcome though.

**Comment by umbreon222** [comment]
Reverted changes to move_to_fort. The only commit that remains is the fix for no pokeball spam.

**Comment by k4n30** [comment]
@douglascamata can this be merged now?

File: pokemongo_bot/cell_workers/pokemon_catch_worker.py [hunk 1]
```diff
@@ -67,6 +67,10 @@ def __init__(self, pokemon, bot, config):
     ############################################################################
 
     def work(self, response_dict=None):
+        pokeballs = self.bot.item_inventory_count(1)
+        superballs = self.bot.item_inventory_count(2)
+        ultraballs = self.bot.item_inventory_count(3)
+
         response_dict = response_dict or self.create_encounter_api_call()
```
Comment (rawgni) [inline pokemongo_bot/cell_workers/pokemon_catch_worker.py]: shouldn't we **NOT** encounter at all if we don't have any pokeballs. I think the stop check should be above this line.

**Pull request merged**